src/allmydata/test/__init__.py

```diff
@@ -52,6 +52,44 @@ def _configure_hypothesis():
     settings.load_profile(profile_name)
 _configure_hypothesis()
 
+def logging_for_pb_listener():
+    """
+    Make Foolscap listen error reports include Listener creation stack
+    information.
+    """
+    from traceback import extract_stack, format_list
+    from foolscap.pb import Listener
+    from twisted.python.log import err
+    from twisted.application import service
+
```

Comment:
Are these imports here instead of at the top for a reason?

Reply:
Cleanliness/compartmentalization I guess.  Kind of like how `_configure_hypothesis` works?  Probably not actually a good reason, I'll move them to the top.